### PR TITLE
Start using ecl output from Ebos

### PR DESCRIPTION
**Pull request by totto82**
Ask ebos to communicate its solution to opm-output directly 

The BlackoilOutputEbos is a stripped down version of SimulatorFullyImplicitBlackoilOutput.hpp
that handles the wells and FIP. My plan is to move this functionality to ebos and remove BlackoilOutputEbos when everything is passed. 

All tests pass and MPI restart works. 
Convergence and results are identical for the big cases. 

Depends on OPM/ewoms#262

### CODE REVIEW

**Comment by andlaus** [comment]
nice!

did you observe any performance change?

**Comment by totto82** [comment]
Yes. The output time is halved for norne but increased (more than noise) for Model 2. I will investigate why.  Maybe you (@andlaus) can keep this in mind while reviewing OPM/ewoms#262?

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this ewoms=262 please

**Comment by totto82** [comment]
These tests pass on my side. Maybe just bad timing with the latest opm-data update.

**Comment by totto82** [comment]
jenkins build this ewoms=262 please

**Comment by totto82** [comment]
Any clue why this fails on Jenkins and not on my side ? Is it related to OPM/opm-data#297? 
I have fresh opm-output and opm-data

**Comment by akva2** [comment]
jenkins build this ewoms=262 please

**Comment by akva2** [comment]
my bad, i had forgotten to force builder for the opm-data job after i added the dedicated post-builder so it updated the wrong workspace. fixed now.

File: opm/autodiff/BlackoilOutputEbos.hpp [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+                    // state used in the computation is actually the one of the last
+                    // step. We need that well state for the gathering. Otherwise
+                    // It an exception with a message like "global state does not
+                    // contain well ..." might be thrown.
+                    // The distribution of data::solution is not done here
+                    data::Solution localCellDataDummy{};
+                    int wellStateStepNumber = ( ! substep && timer.reportStepNum() > 0) ?
+                                (timer.reportStepNum() - 1) : timer.reportStepNum();
+                    // collect all solutions to I/O rank
+                    parallelOutput_->collectToIORank( reservoirStateDummy, localWellState,
```
Comment (blattms) [inline opm/autodiff/BlackoilOutputEbos.hpp]: Please comment on why this is needed here and what exactly it does (as there are a lot of dummies).
I did look at OPM/ewoms#262 and it seems to me that the functionality for gathering the data on the master rank is replicated there and actually used to collect data.  This is triggered in line 171 `ebosSimulator_.problem().writeOutput` and actually done in the EclipseWriter of ewoms. That means that all the maps are there twice and we build them twice. Maybe this is the cause for the performance regression? 

Maybe it is possible to setup and use the data for gathering the data on master only once?

Comment (totto82) [reply opm/autodiff/BlackoilOutputEbos.hpp]: > Please comment on why this is needed here and what exactly it does (as there are a lot of dummies).

From the PR message: 
"The BlackoilOutputEbos is a stripped down version of SimulatorFullyImplicitBlackoilOutput.hpp
that handles the wells and FIP. My plan is to move this functionality to ebos and remove BlackoilOutputEbos when everything is passed."

The reason for the dummies is to be compatible with the legacy code. 

> Maybe this is the cause for the performance regression?

The maps are only made initially (If I have understood the code correctly), It shouldn't be too expensive. But maybe I am wrong.

Comment (totto82) [reply opm/autodiff/BlackoilOutputEbos.hpp]: The reason for the regression was the fact that the ebos code computed some extra output values that was not asked for. Now the output is 3 times faster for both norne and model 2.

**Comment by totto82** [comment]
jenkins build this ewoms=262 please

File: opm/autodiff/BlackoilOutputEbos.hpp (hunk 1 above)
Comment (atgeirr) [reply opm/autodiff/BlackoilOutputEbos.hpp]: Sounds good. I assume that is mostly due to less copying of data?

Comment (totto82) [reply opm/autodiff/BlackoilOutputEbos.hpp]: Yes.  In addition I only supply the output writer with auxiliary values like density, viscosity, bobble point pressures  etc. on reports steps when it is supposed to output. For model 2 output is only every 6 mnd.

**Comment by andlaus** [comment]
I think this is ready. Lacking objections, I'll merge this together with OPM/ewoms#262 in the afternoon.

**Pull request merged**